src/data/finset/lattice.lean

```diff
@@ -249,6 +249,14 @@ le_antisymm
 lemma sup_id_eq_Sup [complete_lattice α] (s : finset α) : s.sup id = Sup s :=
 by simp [Sup_eq_supr, sup_eq_supr]
 
+lemma sup_id_set_eq_sUnion {α} (s : finset (set α)) :
+  s.sup id = ⋃₀(↑s) :=
+sup_id_eq_Sup _
```

Comment:
The suggestion in https://github.com/leanprover-community/mathlib/pull/12459#discussion_r821054916 stilll applies here too.